Sources/StreamChatUI/ChatThread/ChatThreadVC.swift

```diff
@@ -334,8 +352,11 @@ open class ChatThreadVC: _ViewController,
         _ vc: ChatMessageListVC,
         scrollViewDidScroll scrollView: UIScrollView
     ) {
-        // No-op. By default the thread component is not interested in scrollView events,
-        // but you as customer can override this function and provide an implementation.
+        if shouldMarkThreadRead {
```

Comment:
hmm, why here?

Reply:
It is the same logic that we have in `ChatChannelVC`. When the user scrolls, we need to check if we are at the bottom, if yes, we mark it read